timely_beliefs/beliefs/classes.py

```diff
@@ -127,58 +126,45 @@ def source_id(self):
         return None
 
 
-class DBTimedBelief(Base, TimedBelief):
-    """Database representation of TimedBelief"""
+class TimedBeliefDBMixin(TimedBelief):
+    """
+    Mixin class for a table with beliefs.
+    The fields source and sensor do not point to another table - overwrite them to make that happen.
+    """
 
-    __tablename__ = "timed_beliefs"
-    __table_args__ = (
-        UniqueConstraint(
-            "event_start",
-            "belief_horizon",
-            "sensor_id",
-            "source_id",
-            name="_one_belief_by_one_source_uc",
-        ),
-    )
-    # type is useful so we can use polymorphic inheritance
-    # (https://docs.sqlalchemy.org/en/13/orm/inheritance.html#single-table-inheritance)
-    type = Column(String(50), nullable=False)
+    @declared_attr
+    def __table_args__(cls):
+        if has_inherited_table(cls):
+            return (
+                UniqueConstraint(
+                    "event_start",
+                    "belief_horizon",
+                    "sensor_id",
+                    "source_id",
+                    name="_one_belief_by_one_source_uc",
+                ),
+            )
+        return None
 
     event_start = Column(DateTime(timezone=True), primary_key=True)
     belief_horizon = Column(Interval(), nullable=False, primary_key=True)
     cumulative_probability = Column(Float, nullable=False, primary_key=True)
     event_value = Column(Float, nullable=False)
-    sensor_id = Column(
-        Integer(), ForeignKey("sensor.id", ondelete="CASCADE"), primary_key=True
-    )
-    source_id = Column(Integer, ForeignKey("belief_source.id"), primary_key=True)
-    sensor = relationship(
-        "DBSensor",
-        backref=backref(
-            "beliefs", lazy=True, cascade="all, delete-orphan", passive_deletes=True
-        ),
-    )
-    source = relationship(
-        "DBBeliefSource",
-        backref=backref(
-            "beliefs", lazy=True, cascade="all, delete-orphan", passive_deletes=True
-        ),
-    )
 
     @declared_attr
-    def __mapper_args__(self):
-        if self.__name__ == "DBTimedBelief":
-            return {
-                "polymorphic_on": self.type,
-                "polymorphic_identity": "DBTimedBelief",
-            }
-        else:
-            return {"polymorphic_identity": self.__name__}
+    def sensor_id(cls):
+        return Column(
+            Integer(), ForeignKey("sensor.id", ondelete="CASCADE"), primary_key=True
```

Comment:
So when the sensor is deleted, the beliefs are deleted, too? If so, please add that to the docs.